torch/optim/sradam.py

```diff
@@ -0,0 +1,91 @@
+import torch
+from .adam import Adam
+
+
+# TODO(crcrpar): Decide whether to override `state_dict` method as
+# this optimizer tracks square root of `exp_avg_sq` and `max_exp_avg_sq`
+# that contradicts with its parent optimizer :class:`torch.optim.Adam`.
```

Comment:
I think this would be a nice to have to make both optimizers compatible.

Reply:
Naively implemented the feature for the compatibility.

- [SRAdam](https://github.com/csarofeen/pytorch/pull/17/commits/5faf1a96caa3bf458d425a28cdfeecddf15537ba#diff-735fa5dfbcaa19fb4f36ace1f669c428)
- [SRAdamW](https://github.com/csarofeen/pytorch/pull/17/commits/5faf1a96caa3bf458d425a28cdfeecddf15537ba#diff-a5be7ac632ae7779a377f6ac40859580)
- [test](https://github.com/csarofeen/pytorch/pull/17/commits/5faf1a96caa3bf458d425a28cdfeecddf15537ba#diff-d237dd23d765aba7cb30b4c21a5e1a86)